context/context.go

```diff
@@ -104,6 +104,13 @@ func (ctx *Ctx) LoadConfig(path *string) error {
 	if ctx.UserConfig.CollectorConfigurationDirectory == "" {
 		log.Fatal("No collector configuration directory was configured.")
 	}
+	if !filepath.IsAbs(ctx.UserConfig.CollectorConfigurationDirectory) {
+		log.Fatal("Collector configuration directory must be an absolute path.")
+	}
+	err = common.CreatePathToFile(ctx.UserConfig.CollectorConfigurationDirectory)
+	if err != nil {
```

Comment:
This doesn't fail on sidear startup:

```
$ fgrep configuration_directory local.yml 
collector_configuration_directory: "/foo"
$ ./build/1.0.0/linux/amd64/graylog-sidecar -c local.yml 
INFO[0000] Using node-id: 1c4a8275-ffeb-4c27-8cfb-cd23ff5a52e8 
INFO[0000] Starting signal distributor                  
INFO[0015] Adding process runner for: filebeat-6.3.0    
INFO[0015] [filebeat-6.3.0] Configuration change detected, rewriting configuration file. 
INFO[0015] Trying to create directory for: /foo/filebeat-6.3.0.conf 
ERRO[0015] Not able to create directory path: /foo      
```

I see the error only when the sidecar tries to write the config file but the sidecar keeps running.

Reply:
Ah, that's because you've tested it with a directory that was already there (`/`)
I improved the test so it would not ignore the last directory of the path.